app/assets/javascripts/components/student/application/studentApplicationFooter.js.jsx

```diff
@@ -1,14 +1,15 @@
 class StudentApplicationFooter extends React.Component {
 
   validateEssay() {
-    if (this.props.message == 'Unsaved Changes') {
-      return { message: 'Save Before Submitting' }
-    } else if (this.props.essay.length < 1) {
+    const { essay, message } = this.props;
+    if ( essay.length < 1) {
       return { message: 'Essay cannot be blank' }
+    } else if (['Unsaved Changes', 'Save Before Submitting'].includes(message)) {
+      return { message: 'Save Before Submitting' }
     } else {
       return {
         confirm: true,
-        essay: this.props.essay
+        essay
```

Comment:
This change looks good but feels quite odd to be checking against the message copy as opposed to some other more specific piece of state.

Good for now though!

Reply:
Agreed, I was thinking the same thing.  It works for now, but I agree it is definitely something that could be refactored in the future.